gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    """
+    if scipy.sparse.issparse(vec):
```

Comment:
The purpose of `isbow()` is just to check if it is bag of words in the sense that if it is in a 2-tuple format. Jaccard has a different definition for bow kind of inputs, and I can also speed up the other distance functions if I have this info.

I understand that a sparse bag-of-words is not the same format as `scipy.sparse`, but I am not really checking if it is a sparse matrix or not (conceptually) - I am only checking if it happens to be in `scipy.sparse` form so I can convert it to a form (`ndarray`) I can work with.

I can also remove the `scipy.sparse` check here in the `isbow()` and only check for the same in the distance functions itself; I can then change it to dense and then call `isbow()`. That should clear up some confusion, I think.